### PR TITLE
Story/add custom functions art 1947

### PR DESCRIPTION
**Pull request by archan937**
### Add custom functions

Give the pro-coder the ability to push custom functions to their Betty applications.

The following commands will be added:

* `bb functions init <identifier>`
* `bb functions build`
* `bb functions publish`

The custom functions will be available in the new actions runtime. A defined (custom Javascript) function is able as an event.

### CODE REVIEW

File: assets/functions/packer/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "scripts": {
+    "build": "npx webpack --config webpack.config.js"
```
Comment (jongbelegen) [inline assets/functions/packer/package.json:3]: npx feels npm specific, what about yarn users? 

wouldn't it be better to remove npx here and use the local dep?

File: package.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
   "scripts": {
     "build": "tsc",
     "commit": "commit",
+    "link": "yarn unlink; yarn build; yarn link",
```
Comment (jongbelegen) [inline package.json:15]: is this only being used for dev on this package and not expected to be used by the end user?

Comment (archan937) [reply package.json:15]: Yeah, you're right.

File: assets/functions/packer/package.json (hunk 1 above)
Comment (archan937) [reply assets/functions/packer/package.json:3]: Yeah, fair point. The packer project is an isolated project which is "invisible" for the pro-coder. So in this case, it is acceptable.

**Pull request merged**